router/router_test.go

```diff
@@ -108,3 +104,129 @@ func TestRingpopRouterMakeRemoteClientCached(t *testing.T) {
 	router.GetClient("hello")
 	cf.AssertNumberOfCalls(t, "MakeRemoteClient", 1)
 }
+
+func TestRingpopRouterRemoveExistingClient(t *testing.T) {
+	rp := &mocks.Ringpop{}
+	rp.On("RegisterListener", mock.Anything).Return()
+
+	router := New(rp, nil, nil).(*router)
+
+	hostport := "127.0.0.1:3000"
+	router.clientCache[hostport] = "Some client"
+	assert.Equal(t, "Some client", router.clientCache[hostport])
```

Comment:
You are completely right, this assert is useless. But as discussed on an other line I will rewrite how the cache is populated to not tamper with the internals of the router during tests. So this assert will become a necessity.